src/Database/Connection.ts

```diff
@@ -135,13 +135,10 @@ export class Connection extends EventEmitter implements ConnectionContract {
     options?: TransactionOptions,
   ): Promise<TResult> {
     const db = await this._ensureDb();
-    let result: TResult;
-    await this.client.withSession(async (session) => {
+    return this.client.withSession(async (session) => {
       return session.withTransaction(async (session) => {
-        result = await handler(session, db);
+        return handler(session, db);
```

Comment:
I'm afraid of this change. Pretty sure `await` was here to make sure the exception happens in the right scope

Reply:
I'll revert so